### ISSUE TITLE
Improvements to the screen during navigation

### ISSUE THREAD

**Issue by ooosssay** [organicmaps/organicmaps#8267]
Let me start by saying that to be honest all of these requests come from existing features from Apple Maps but I decided to put them here as I believe it will help with the current navigation and hopefully make this app one of the best.

* While driving the route should usually stay in view rather than having the screen straight where you’re pointed at. This can help show more of the route than some of it. Also, note that Apple Maps still somewhat shows the direction where you’re driving so somewhere in between the route and where you’re pointed at.

![IMG_6651](https://github.com/organicmaps/organicmaps/assets/129241633/1d185585-913f-4b62-b010-348aafc5d139)

* Having a lot of road names can be overwhelming to the user so I’ve noticed 3 things that Apple Maps does to help combat this.

    1. The current road name that the user is driving on is shown at the bottom of the screen on top of the road with the same color (blue) rather than on the road.

![IMG_6651](https://github.com/organicmaps/organicmaps/assets/129241633/21712f50-1e32-4158-91d1-2e34d4d6fd12)
![IMG_6653](https://github.com/organicmaps/organicmaps/assets/129241633/794c66fa-5ec7-4990-b5ea-87b6bb31a77e)

2. The names of the roads that are connected to the road that the user is driving on are shown with popouts beside the current road (I forgot what they are called). This can help the user read the road names horizontally rather than vertically and also helps the visibility of the name when 3D objects are nearby.

![IMG_6649](https://github.com/organicmaps/organicmaps/assets/129241633/6cb970a0-ae32-476f-98ea-e053cedd07b1)

3. Show only the road names that the user can go on rather than all the roads in view, but when the user uncenters the map it shows all the names of the roads.

![IMG_6649](https://github.com/organicmaps/organicmaps/assets/129241633/5f0ea568-22c7-4937-bf91-b52d131febf0)

* When the user is coming up to a turn the map automatically goes from the perspective view to a top-down view. For example, when approaching a turn the map would try to center between the road the user is currently on and the road the user is about to head into so it shows where the user is heading. This helps the user to actually see the turn with its nearby features and also helps when there’s a tall 3d building (New York). Note that the map isn’t centered directly on the turn.

![IMG_6653](https://github.com/organicmaps/organicmaps/assets/129241633/f45a91cd-5601-4b2e-91c1-f1825fabfd34)

* I’ve noted that the way Apple Maps decides to either zoom in or out can depend on the features near you. For example, when there are many roads and buildings (towns and cities and neighborhoods [all of these places have a lot of things going on like pedestrians and traffic lights and changing road lanes]) near you Apple Maps decides to zoom in and stay there until once Apple Maps notices that you’re in a more feature less area (Major highways enclosed by forests, roads in forests, open landscapes that can consist of farmland or simply grass) then Apple Maps decides to zoom out.

**Comment by matheusgomesms** [comment]
There's plenty of UI discussion here, but this adds a lot to the discussion. In my opinion, Apple Maps has today the best driving mode in navigation of all apps.

**Comment by grahamjpark** [comment]
I came here to give similar feedback. Let me know if this doesn't belong here and should be pulled to separate issues.

Issue 1: Zoom

I just used organic maps while driving for the first time during my holiday. I found that it was too zoomed out most of the time. There were two instances where the road had a bend, but on screen it looked like the line was straight because of how zoomed out it was. I even made note of the locations to edit them in OSM before I realized it.

It also took some getting used too. Both my wife and I mistakenly thought a turn was coming up because it's placement on the screen way too far in advance.

Another example I screenshotted below. There's no value in displaying so far beyond the turn IMO.

![Screenshot_20240613-172531.png](https://github.com/organicmaps/organicmaps/assets/5332575/3b5c1c1a-3752-401d-90d4-564e80d5d524)

Issue 2: Parking prioritization

The UI prioritized parking icons way too much. This screenshot only occured when slowing down on a highway. It normally wouldn't be this comical, but there were times it prioritized showing a business's parking lot instead of the business itself. Knowing that my turn is after Starbucks is much more useful than knowing there are 4 parking lots at the intersection.

![Screenshot_20240604-150509.png](https://github.com/organicmaps/organicmaps/assets/5332575/79167943-4548-408d-b9f8-dddc9e00fea2)

Issue 3: Next turn name font too small

Without the audio cues to read the road names, the information about my next turn at the top of the screen was dangerously small. Making me squint and study the screen while driving.

Overall excited by Organic Maps though and was happy to test out driving navigation!

**Comment by biodranik** [comment]
@grahamjpark thanks! Can you please clarify:

> Both my wife and I mistakenly thought a turn was coming up because its placement on the screen way too far in advance.

Does it mean that the turn was announced many km in advance, and it wasn't clear? Or was it something else?

**Comment by ooosssay** [comment]
@biodranik @grahamjpark  I think that he meant that the screen was really zoomed out that it looked like a turn was right in front of them but in reality they had some distance to the turn.

**Comment by grahamjpark** [comment]
Yep, that's right @ooosssay. We saw it appear on the screen and assumed the distance based on that.

We probably either didn't listen for the audio cue or we ignored it because we're still transitioning from using miles to kilometers (we just moved). For whatever reason we looked to the screen and saw the turn coming up and thought it was much closer.

**Comment by Maschga** [comment]
I also have the problem that the app zooms out too far, so I can't see exactly where I need to go.
It would be great to display the zoom level depending on the distance to the next turn shown at the top left.

**Comment by RedAuburn** [comment]
Guru maps has a really nice horizontal ui, that saves a lot of space. IMO OM should have something like this:

![Screenshot_20241206_213706_Guru_Maps.jpg](https://github.com/user-attachments/assets/dbf0ff87-2405-469f-a28f-73fc94d3f4d8)

**Comment by alensiljak** [comment]
> Guru maps has a really nice horizontal ui, that saves a lot of space. IMO OM should have something like this:

Somehow I never liked horizontal navigation. I'm driving in the up/down direction and am more interested in what's ahead than what's 3km to the side. Unless I'm trying to get out of a one-way maze, that is.
